### DISCUSSION TITLE
How to use `$get(...)` when referring to a `MorphToSelect::` field

### DISCUSSION THREAD

**Discussion by nexxai** [filamentphp/filament#14507] [Help]
### Package

Form builder

### Package Version

v3.2.110

### How can we help you?

I've got a form where I want an element to only appear when the value of a `MorphToSelect::` field above it matches a certain criteria.

```php
MorphToSelect::make('entrant')
  ->label('Entrant')
  ->types(function (RelationManager $livewire) {
      $contest = $livewire->getOwnerRecord();

      $fields = [];

      $fields[] = MorphToSelect\Type::make(User::class)
          ->label('Members')
          ->titleAttribute('name');

      $fields[] = MorphToSelect\Type::make(AnonymousUser::class)
          ->label('Non-members')
          ->titleAttribute('name');

      return $fields;
  }),
```

I only want the `referral_source_id` field to appear when the selected `entrant` is an `AnonymousUser`.

I've tried every permutation of this:

```php
Select::make('referral_source_id')
  ->label('Referral')
  ->relationship('referralSource', 'name')
  ->visible(function (Get $get) {
      return $get('entrant') instanceof AnonymousUser;                                  <- Tried this
      return AnonymousUser::where('id', $get('entrant')->id)->exists();                 <- And this
      return AnonymousUser::find($get('entrant'));                                      <- This too
  })
```

But no matter what I do, `$get('entrant')` is always `null` it obviously never shows.

Am I missing something or is what I'm trying to do impossible because it's a Morph model instead of a regular relationship?

**Comment by wbpols** [comment]
I am experiencing something similar, although in my case it is the other way around (trying to access something from the 'main' form inside the `->types()` function).

Anyway, maybe this will help you: if you do a `dd($get('*'));` within the `Select::make('referral_source_id')->visible()` function you can see all values on the current form.

**Reply by wbpols** [reply]
I just stumbled upon [this in the documentation](https://filamentphp.com/docs/3.x/forms/fields/repeater#using-get-to-access-parent-field-values) which helped me continue, perhaps it will help you as well @nexxai.

**Reply by yusufkandemir** [reply]
@wbpols thank you! `$get('*')` doesn't show a meaningful output, but `$get('../something')` works! I think it should not be buried into the repeater docs, as it applies to any nested form.

**Comment by micraux** [comment]
Hello,

@nexxai: did you find a way to do what you expected?

I am trying to do the same: retrieve the state of a `MorphToSelect` field...

**Reply by nexxai** [reply]
Unfortunately, no, I never found a solution.  Sorry :(

**Reply by micraux** [reply]
Well, I found a solution using the `$get('*')` mentioned above.

I don't know if it is the best way to do it but it works for me...

Oh, and I discovered too that you have to use `->live(onBlur: true)` in order to make it work properly.

As explained, I need to retrieve the state of a `MorphToSelect` field, which is not exactly what you want to do, but I think you can adapt it to your needs.

I put my code here just in case...

```php
->live(onBlur: true)
->afterStateUpdated(
    function ($state, Forms\Get $get, Forms\Set $set) {
        $gets = $get('*');
        $article = resolve($gets[0])::find($gets[1]);
        $set('price', $article->price);
        return;
    }
)
```

**Comment by peteleco** [comment]
Hi everyone, i also bump into this, after finding this discussion i was able to discover a way to get the correct value debuging using the $get('*'), so to get the value in your example just add the suffix "_type" assuming that you are using in the table entrant_type/entrant_id

```php
->visible(function (Get $get) {
      return $get('entrant_type') instanceof AnonymousUser;
      return $get('entrant_type') === Relation::getMorphAlias(AnonymousUser::class); // If you are using alias
  })
```

```php
\Illuminate\Database\Eloquent\Relations\MorphTo::getMorphType:74
$typeColumn = $relationship->getMorphType();

\Illuminate\Database\Eloquent\Relations\MorphTo::getMorphType:83
$typeSelect = Select::make($typeColumn)

```
